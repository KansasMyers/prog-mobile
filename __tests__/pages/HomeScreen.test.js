import React from "react";
import renderer from 'react-test-renderer';
import HomeScreen from "../../src/pages/HomeScreen";

// Mockando o componente MyImageButton pois ele não é o objeto de teste nesse caso.
jest.mock('../../src/pages/components/MyImageButton', () => 'MyImageButton');

// Criando os props mockados para o teste
const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

describe("<HomeScreen />", () => {
  var props = createTestProps({});

  it('renders correctly', () => {
    const tree = renderer.create(<HomeScreen navigation={props.navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has 1 child", () => {
    const tree = renderer.create(<HomeScreen navigation={props.navigation} />).toJSON();
    expect(tree.children.length).toBe(1);
  });
});