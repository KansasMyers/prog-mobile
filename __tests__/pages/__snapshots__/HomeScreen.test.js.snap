// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <MyImageButton
          btnColor="#00F2E0"
          btnIcon="user-plus"
          customClick={[Function]}
          title="Cadastrar Usuário"
        />
        <MyImageButton
          btnColor="#A45BB9"
          btnIcon="user-circle"
          customClick={[Function]}
          title="Atualizar Usuário"
        />
        <MyImageButton
          btnColor="#F9AD29"
          btnIcon="user"
          customClick={[Function]}
          title="Visualizar Usuário"
        />
        <MyImageButton
          btnColor="#384F62"
          btnIcon="users"
          customClick={[Function]}
          title="Visualizar Todos os Usuários"
        />
        <MyImageButton
          btnColor="#D1503A"
          btnIcon="user-times"
          customClick={[Function]}
          title="Excluir Usuário"
        />
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
